core/src/main/java/org/elasticsearch/cluster/metadata/IndexMetaData.java

```diff
@@ -693,6 +735,25 @@ public Custom getCustom(String type) {
             return this.customs.get(type);
         }
 
+        public Builder putActiveAllocationIds(int shardId, AllocationIds allocationIds) {
+            activeAllocationIds.put(shardId, allocationIds);
+            return this;
+        }
+
+        public Builder removeActiveAllocationIds(int shardId) {
+            activeAllocationIds.remove(shardId);
+            return this;
+        }
+
+        public Builder removeAllActiveAllocationIds() {
+            activeAllocationIds.clear();
+            return this;
+        }
+
+        public AllocationIds getActiveAllocationIds(int shardId) {
```

Comment:
It would be great, if we move to a DiffableStringSet, to make this version return a Set<String> and bellow the hood return an immutable set.